### DISCUSSION TITLE
Hierarchical data

### DISCUSSION THREAD

**Discussion by mollykreis** [perspective-dev/perspective#1996] [Q&A]
I have data that represents a hierarchy, and I'd like to be able to represent this data in Perspective. Specifically, I'm looking to create a view config that will allow me to query my data and return rows ordered by their parent.

I was thinking about a schema that looks something like this but am open to other schemas if it would make the task at hand easier:
```js
{
    id: 'string',         // This is the unique index for my data
    hasChildren: 'boolean',
    parentId: 'string'    // This is the id of the parent, or null/empty if there is no parent

    // Other data here
}
```

I'd like to create a view where I data is ordered as shown below so that I have the necessary rows to draw a table based on a virtualized start_row and end_row. It would also be ideal if I could leverage Perspective's concept of [expanding](https://perspective.finos.org/docs/obj/perspective#viewexpand---promise-void) and [collapsing](https://perspective.finos.org/docs/obj/perspective#viewcollapse---promise-void) so that if I collapse part of my hierarchical table, the view could take that into account when I request data from it.
- top-level item 1
	- child 1
	- child 2
		- grandchild 1
		- grandchild 2
- top-level item 2 (no children)
- top-level item 3
	- child 1
	- child 2
	- child 3

Are there any recommendations about the best way to go about this using Perspective? Is there a better way to model my data to make this easier?

**Comment by rajsite** [comment]
@texodus would appreciate it if there is any insight!

We are looking at making a custom visualization plugin for perspective (a new table with some different features) but would have the requirement of handling tree / hierarchy style data (i.e. the id relationships @mollykreis mentioned) and haven't figured out the best way to create Perspective Views on hierarchy style of data.
